src/System.Drawing.Common/src/System/Drawing/Text/PrivateFontCollection.cs

```diff
@@ -57,13 +57,38 @@ protected override void Dispose(bool disposing)
         /// </summary>
         public void AddFontFile(string filename)
         {
-            Path.GetFullPath(filename);
+            if (_nativeFontCollection == IntPtr.Zero)
+            {
+                // This is the default behavior on Desktop. The ArgumentException originates from GdipPrivateAddFontFile which would
+                // refuse the null pointer.
+                throw new ArgumentException();
+            }
+
+            if (filename == null)
+            {
+                // This is the default behavior on Desktop. The name "path" originates from Path.GetFullPath or similar which would refuse
+                // a null value.
+                throw new ArgumentNullException("path");
+            }
+
+            // this ensure the filename is valid (or throw the correct exception)
+            string fullPath = Path.GetFullPath(filename);
 
-            int status = SafeNativeMethods.Gdip.GdipPrivateAddFontFile(new HandleRef(this, _nativeFontCollection), filename);
+            if (!File.Exists(fullPath))
+            {
+                throw new FileNotFoundException();
+            }
+
+            int status = SafeNativeMethods.Gdip.GdipPrivateAddFontFile(new HandleRef(this, _nativeFontCollection), fullPath);
             SafeNativeMethods.Gdip.CheckStatus(status);
 
             // Register private font with GDI as well so pure GDI-based controls (TextBox, Button for instance) can access it.
-            SafeNativeMethods.AddFontFile(filename);
+            // Do this on Windows only as Unix only has GDI+ (libgdiplus), not GDI; and we don't have System.Windows.Forms
+            // on Unix.
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Let's make the same change here as in the other PR: split into Windows+Unix partial classes and have compile-time logic differences rather than runtime.